core/cloudflow-cr-generator/src/main/scala/cloudflow/cr/Generator.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package cloudflow.cr
+
+import com.typesafe.config.{ ConfigFactory, ConfigRenderOptions }
+
+import java.io.File
+import java.net.URL
+import scala.collection.JavaConverters._
+import scala.io.Source
+
+import scopt.OParser
+import spray.json._
+import cloudflow.blueprint._
+import cloudflow.blueprint.StreamletDescriptorFormat._
+import cloudflow.blueprint.deployment._
+import cloudflow.extractor._
+
+object Generator {
+
+  val cloudflowVersion = BuildInfo.version
+
+  private def makeCR(appDescriptor: ApplicationDescriptor, version: String): CloudflowCR =
+    CloudflowCR(
+      apiVersion = "cloudflow.lightbend.com/v1alpha1",
+      kind = "CloudflowApplication",
+      metadata = Metadata(
+        annotations = Map("com.lightbend.cloudflow/created-by-cli-version" -> version),
+        labels = Map(
+          "app.kubernetes.io/managed-by" -> "cloudflow",
+          "app.kubernetes.io/part-of" -> appDescriptor.appId,
+          "com.lightbend.cloudflow/app-id" -> appDescriptor.appId),
+        name = appDescriptor.appId),
+      appDescriptor)
+
+  def generate(
+      projectId: String,
+      version: String,
+      blueprintStr: String,
+      classpath: List[String],
+      dockerImages: String => String): String = {
+    val scanConfig =
+      DescriptorExtractor.ScanConfiguration(projectId = projectId, classpathUrls = classpath.map(new URL(_)).toArray)
+
+    val extractedStreamlets = DescriptorExtractor.scan(scanConfig)
+
+    val streamletDescriptors = extractedStreamlets.map {
+      case (_, configDescriptor) =>
+        configDescriptor
+          .root()
+          .render(ConfigRenderOptions.concise())
+          .parseJson
+          .convertTo[cloudflow.blueprint.StreamletDescriptor]
+    }
+
+    val blueprintConfig = ConfigFactory.parseString(blueprintStr).resolve()
+
+    val blueprint = Blueprint.parseConfig(blueprintConfig, streamletDescriptors.toVector)
+    if (blueprint.problems.isEmpty) {
+      val appDescriptor = {
+        val appVersion = version
+        val agentPathsMap = Map("prometheus" -> "/prometheus/jmx_prometheus_javaagent.jar")
+        val libraryVersion = cloudflowVersion
+
+        ApplicationDescriptor(
+          appId = projectId,
+          appVersion = version,
+          (k) => dockerImages(k),
+          blueprint.verified.right.get,
+          agentPathsMap,
+          cloudflowVersion)
+      }
+
+      import cloudflow.blueprint.deployment.CloudflowCRFormat._
+      val cloudflowCr = makeCR(appDescriptor, version)
+      cloudflowCr.toJson.compactPrint
+    } else {
+      throw new Exception(s"Found errors: ${blueprint.problems.mkString}")
+    }
+  }
+
+  case class Config(
+      name: String = null,
```

Comment:
nitpick: can make `final case class Config`. Purely my personal choice - I try to make everything `final` unless forced otherwise. But feel free to ignore :-)

Reply:
done